src/utils/daos_hdlr.c

```diff
@@ -49,12 +49,18 @@ struct file_dfs {
 };
 
 /* Report an error with a system error number using a standard output format */
-#define DH_PERROR_SYS(AP, RC, STR, ...)					\
-	fprintf((AP)->errstream, STR ": %s (%d)\n", ## __VA_ARGS__, strerror(RC), (RC))
+#define DH_PERROR_SYS(AP, RC, STR, ...)                                                            \
+	do {                                                                                       \
+		fprintf((AP)->errstream, STR ": %s (%d)\n", ##__VA_ARGS__, strerror(RC), (RC));    \
+		fflush((AP)->errstream);                                                           \
+	} while (0)
```

Comment:
@tanabarr @mjmac without this flush I was seeing that sometimes this output was being lost.  This PR introduces changes to check certain things are present in the output which is now I found this.

That makes this part of the PR a bug fix and therefore something we want on 2.4.

Reply:
Maybe related to
https://daosio.atlassian.net/browse/DAOS-11551
And also maybe part of the fix needed for https://daosio.atlassian.net/browse/DAOS-13499